index.js

```diff
@@ -44,7 +46,15 @@ async function run() {
 
 		// Get commits between computed range
 		let {stdout: commits} = await execFile('git', ['log', '--format=%H%s', range]);
-		commits = commits.split('\n').filter(Boolean);
+		commits = commits.split('\n').filter(Boolean).map(line => ({
+			hash: includeHash ? line.slice(0, 8) : '',
+			title: line.slice(40)
+		}));
+
+		if (exclude) {
+			const regex = new RegExp(exclude);
+			commits.filter(({title}) => !regex.test(title));
+		}
```

Comment:
Beware that `regex.test()` with any string passed as argument will return `true` if `exclude === ''`, but that would be handled by `if (exclude) ...` anyway.